src/api/server.rs

```diff
@@ -140,16 +140,24 @@ fn make_routes(
     // Handle GraphQL queries. Headers will first be parsed to determine whether the query is
     // a subscription and if so, an attempt will be made to upgrade the connection to WebSockets.
     // All other queries will fall back to the default HTTP handler.
-    let graphql_handler = warp::path("graphql").and(graphql_subscription_handler.or(
-        async_graphql_warp::graphql(schema::build_schema().finish()).and_then(
-            |(schema, request): (Schema<_, _, _>, Request)| async move {
-                Ok::<_, Infallible>(GraphQLResponse::from(schema.execute(request).await))
-            },
-        ),
-    ));
+    let graphql_handler = if api.graphql {
+        warp::path("graphql")
+            .and(graphql_subscription_handler.or(
+                async_graphql_warp::graphql(schema::build_schema().finish()).and_then(
+                    |(schema, request): (Schema<_, _, _>, Request)| async move {
+                        Ok::<_, Infallible>(GraphQLResponse::from(schema.execute(request).await))
+                    },
+                ),
+            ))
+            .boxed()
+    } else {
+        warp::any()
+            .and_then(|| async { Err(warp::reject::not_found()) })
+            .boxed()
+    };
 
     // Provide a playground for executing GraphQL queries/mutations/subscriptions.
-    let graphql_playground = if playground {
+    let graphql_playground = if api.playground & api.graphql {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** I think the logical operator was meant to be used here instead of the binary operator‏

```suggestion
    let graphql_playground = if api.playground && api.graphql {
```